lib/pbench/agent/results.py

```diff
@@ -117,13 +122,13 @@ def make_result_tb(self):
             except OSError as exc:
                 if exc.errno != errno.ENOENT:
                     self.logger.error("error removing failed tar ball, {}", tarball)
-            sys.exit(1)
+            return (1, "")
 
         self.make_md5sum(tarball)
 
         # The contract with the caller is to just return the full path to the
         # created tar ball.
-        return str(tarball)
+        return (0, str(tarball))
```

Comment:
Seems that it would be simpler and equally informative to simply return the tarball name or `None` on failure since the pattern is always `(0, str(tarball))` or `(1, "")`. Although I'd be in favor of reporting failure using a named exception; exceptions in an API have the advantage that they're harder to accidentally ignore, and the meaning is more clear.

Reply:
yeah right